src/core/lib/event_engine/posix_engine/posix_endpoint.cc

```diff
@@ -1132,7 +1165,12 @@ void PosixEndpointImpl::Write(
     current_zerocopy_send_ = zerocopy_send_record;
     handle_->NotifyOnWrite(on_write_);
   } else {
-    on_writable(status);
+    // TODO(vigneshbabu): Consider eventually running this callback inline to
+    // avoid a thread hop. At the time of submission, it causes deadlocks which
+    // should be reolved after ExecCtx removal.
+    engine_->Run([on_writable = std::move(on_writable), status]() mutable {
+      on_writable(status);
```

Comment:
Unless you're 100% sure it isn't necessary, EE callbacks should begin with local defines of ExecCtx and ApplicationExecCtx variables. They aren't guaranteed to exist otherwise.

Reply:
I think rather than taking a dependency on ExecCtx here, we can enforce inside core that the on_write callback should contain an ExecCtx and an ApplicationExecCtx within the callback body. This would be similar to how were are migrating the RunAfter timers.

Reply:
I think that could work. We can stipulate that any non-EventEngine-internal uses of the following within core will need to instantiate their own ExecCtx and ApplicationCallbackExecCtx within the callback:

* `::Run`
* `::RunAfter`
* `::Connect`'s OnConnectCallback
* `::CreateListener`'s AcceptCallback and the on_shutdown callback
* `::DNSResolver::Lookup*` callbacks
* `::Endpoint::Read` and `::Endpoint::Write` callbacks.